ext/php7/ddtrace.c

```diff
@@ -251,24 +259,50 @@ static void dd_register_fatal_error_ce(TSRMLS_D) {
     ddtrace_ce_fatal_error = zend_register_internal_class_ex(&ce, zend_ce_exception TSRMLS_CC);
 }
 
-static void _dd_disable_if_incompatible_sapi_detected(TSRMLS_D) {
-    if (strcmp("fpm-fcgi", sapi_module.name) == 0 || strcmp("apache2handler", sapi_module.name) == 0 ||
-        strcmp("cli", sapi_module.name) == 0 || strcmp("cli-server", sapi_module.name) == 0 ||
-        strcmp("cgi-fcgi", sapi_module.name) == 0) {
+/*
+ * Tells whether or not the script is executed from the command line.
+ */
+static bool dd_is_cli() { return strcmp("cli", sapi_module.name) == 0; }
+
+static void dd_disable_if_incompatible_sapi_detected(TSRMLS_D) {
+    if (strcmp("fpm-fcgi", sapi_module.name) == 0 || strcmp("apache2handler", sapi_module.name) == 0 || dd_is_cli() ||
+        strcmp("cli-server", sapi_module.name) == 0 || strcmp("cgi-fcgi", sapi_module.name) == 0) {
         return;
     }
     ddtrace_log_debugf("Incompatible SAPI detected '%s'; disabling ddtrace", sapi_module.name);
     DDTRACE_G(disable) = 1;
 }
 
+/*
+ * Disable tracing if user explicitly disabled tracer by configuration or if a CLI SAPI is used and CLI tracing
+ * is not explictly enabled.
+ */
+static void dd_disable_by_configuration(TSRMLS_D) {
+    if (!get_dd_trace_enabled()) {
+        ddtrace_log_debugf("Tracing is disabled by user setting; disabling ddtrace");
+        DDTRACE_G(disable) = 1;
+        return;
+    }
+
+    if (dd_is_cli() && !get_dd_trace_cli_enabled()) {
+        ddtrace_log_debugf("CLI SAPI not explicitly enabled; disabling ddtrace");
+        DDTRACE_G(disable) = 1;
+        return;
+    }
+}
+
 static PHP_MINIT_FUNCTION(ddtrace) {
     UNUSED(type);
     REGISTER_STRING_CONSTANT("DD_TRACE_VERSION", PHP_DDTRACE_VERSION, CONST_CS | CONST_PERSISTENT);
     REGISTER_INI_ENTRIES();
 
     // config initialization needs to be at the top
     ddtrace_initialize_config(TSRMLS_C);
-    _dd_disable_if_incompatible_sapi_detected(TSRMLS_C);
+
+    dd_disable_by_configuration(TSRMLS_C);
```

Comment:
I don't think this will work as expected on the FPM and Apache SAPIs. Those SAPIs don't have access to the environment vars until the [first RINIT](https://github.com/DataDog/dd-trace-php/blob/f09dd36/ext/php7/ddtrace.c#L351-L355). INI settings do not have this same restriction so they can be reliably accessed in MINIT/startup for all SAPIs.

What are your thoughts on deprecating `DD_TRACE_ENABLED` and `DD_TRACE_CLI_ENABLED` in favor of exclusively using INI settings to disable the tracer (e.g. `ddtrace.disable=1`)? On the CLI SAPI `ddtrace.disable` could default to `1`, and all other SAPIs the default would be `0`.